File: solutions/LastFactorialDigit.java
import java.util.Arrays;
import java.util.Scanner;

public class LastFactorialDigit {
    private static int factorial(int n) {
        if(n == 1) {
            return n;
        } else {
            return n * factorial(n - 1);
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        int[] tests = new int[t];

        for(int i = 0; i < t; i++) {
            int n = sc.nextInt();
            tests[i] = factorial(n)%10;
        }

        sc.close();

        Arrays.stream(tests).forEach(System.out::println);
    }
}

File: solutions/last_factorial_digit.py
t = int(input())
tests = []


def fact(n):
    if n == 1:
        return n
    else:
        return n * fact(n - 1)


while t > 0:
    n = int(input())
    tests.append(fact(n))
    t -= 1

for x in tests:
    print(x % 10)